### PR TITLE
Add item pickup popup

### PR DESCRIPTION
**Pull request by samuelWilliams99**
image
Popups move in from the right, stack up on top of each other, then fade out after 4 seconds

### CODE REVIEW

File: gamemode/modules/inventory/cl_pickup_popup.lua [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+                local xOffsetProg = 1 - ( 10 * math.Clamp( timePassed, 0, 0.1 ) )
+                local popupX = x + xOffsetProg * popupWidth
+
+                inv.drawPickupPopup( popup, popupX, y, popupWidth, popupHeight, fadeProg * 255 )
+            end
+        end
+
+    render.SetScissorRect( 0, 0, 0, 0, false )
+end )
```
Comment (overlordbleck) [inline gamemode/modules/inventory/cl_pickup_popup.lua]: new linee

File: gamemode/modules/inventory/cl_pickup_popup.lua [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+local popupItemMaterial = Material( "rvr/backgrounds/craftingmenu_ingredient.png" )
+
+local popupDuration = 4
+local brown = Color( 91, 56, 34 )
+
+net.Receive( "RVR_Inventory_OnPickup", function()
+    local itemData = net.ReadTable()
+    local count = net.ReadUInt( 16 )
+
+    table.insert( inv.pickupPopups, 1, { itemData = itemData, count = count, startTime = CurTime(), iconMat = Material( itemData.icon ) } )
```
Comment (overlordbleck) [inline gamemode/modules/inventory/cl_pickup_popup.lua]: multiline with that table?

File: gamemode/modules/inventory/cl_pickup_popup.lua [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    surface.SetDrawColor( brown.r, brown.g, brown.b, alpha )
+    surface.DrawRect( nameX, y + h * 0.72, ( x + w ) - nameX - 10, 2 )
+end
+
+hook.Add( "HUDPaint", "RVR_Inventory_OnPickup", function()
+    local x = ScrW() - popupWidth
+
+    local cTime = CurTime()
+    if prevTime == 0 then prevTime = cTime end
```
Comment (overlordbleck) [inline gamemode/modules/inventory/cl_pickup_popup.lua]: I'm not really a fan of a conditional statement with an assignment operation on one line. (*up for debate of course*)

Comment (overlordbleck) [reply gamemode/modules/inventory/cl_pickup_popup.lua]: That being said you did it on line 85 correctly

**Review by ghost — APPROVED** [review]
Looks good to me

**Pull request merged**